Use types from k8s.io/metrics

DirectXMan12 opened the pull request:
This PR changes Heapster over to the types from k8s.io/metrics when serving metrics from the API server.

DirectXMan12 committed:
Update Godeps

DirectXMan12 committed:
Remove Monasca Sink

DirectXMan12 committed:
[WIP] Rebase onto Latest Kubernetes

This commit contains the Heapster code changes required to rebase
Heapster onto the latest Kubernetes.  This begins to move away from
making use of `k8s.io/kubernetes` proper, instead favoring
`k8s.io/client-go`, `k8s.io/apimachinery`, and `k8s.io/apiserver`.
This should make future updates easier.

DirectXMan12 committed:
[WIP] Rebase API server code

This commit continues the work of the previous commit, rebasing
the generic-API-server-based code in Heapster to use the latest
copy of generic API server, and to follow the proper conventions,
including switching over to just use webhook authentication.

DirectXMan12 committed:
Remove Summary Fallback

Heapster doesn't work with versions of Kubernetes that don't have
the summary API anymore, so we can remove the version detection and
fallback logic.

DirectXMan12 committed:
Copy parts of Kubelet Client In-Tree

We only need a small part of the Kubelet client which just wraps
the normal REST client config.  By doing this, we can avoid vendoring
much more of Kubernetes.

With this, the only remaining use of the vendored Kubernetes is the
Kubelet stats API.

DirectXMan12 committed:
Use metric types from k8s.io/metrics

This commit switches over the API server to use the metrics API
types from k8s.io/metrics.  It leaves the legacy APIs using the built-in
types.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/heapster/1537)
<!-- Reviewable:end -->

DirectXMan12 committed:
Add k8s.io/metrics to Godeps

This commit adds k8s.io/metrics to Godeps/the vendor directory,
as needed by the previous commit.

DirectXMan12 commented:
cc @piosz PTAL

DirectXMan12 commented:
client-go finally synced, so this is unblocked

piosz commented:
LGTM, thanks a lot for doing it!
